apis/python/src/tiledbsoma/soma_array.cc

```diff
@@ -0,0 +1,506 @@
+                const std::string&, const std::vector<double>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_float32",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<float>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_int64",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<int64_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_int32",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<int32_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_int16",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<int16_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_int8",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<int8_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_uint64",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<uint64_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_uint32",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<uint32_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_uint16",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<uint16_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        .def(
+            "set_dim_points_uint8",
+            static_cast<void (SOMAArray::*)(
+                const std::string&, const std::vector<uint8_t>&)>(
+                &SOMAArray::set_dim_points))
+
+        // In an initial version of this file we had `set_dim_ranges` relying
+        // solely on type-overloading. This worked since we supported only int
+        // and string indices. In a subsequent version we are now supporting
+        // various NumPy/PyArrow types including float32, float64, int8, uint16,
+        // etc. It is an unfortunate fact that pybind11 does _not_ successfully
+        // disambiguate between float32 and float64, or between int8 and int64,
+        // etc. given that we ask it to disambiguate using not just types but
+        // std::vector of types or std::vector of std::pair of types.
+        // Experiments have shown that when both float32 and float64 are
+        // implemented with overloaded names to be differentiated solely by
+        // type, pybind11 uses the _first found_. Therefore it is necessary for
+        // us to no longer use common overloaded names.
```

Comment:
Thanks for leaving this in

That was the result of a few long days of hard debugging ... very well-earned wisdom. :)